apps/geo/views.py

```diff
@@ -96,6 +99,23 @@ def get_intersects(self, request, pk=None, version=None):
             for id, geom in geoms
         ])
 
+    @action(
+        detail=True,
+        url_path='published',
+        methods=['post'],
+        serializer_class = RegionSerializer,
+        permission_classes=[permissions.IsAuthenticated]
+    )
+    def get_published(self, request, pk=None, version=None):
+        region = self.get_object()
+        if not Region.objects.filter(id=region.id, created_by=request.user):
+            raise ValidationError('Can be published by user who created it')
+
+        region.is_published = True
+        region.save()
```

Comment:
Good habit to use updated_fields when we are changing single field only.

```suggestion
        region.save(update_fields=['is_published'])
```